### PR TITLE
ci: validate CNS in-memory state after restart

### PR DESCRIPTION
**Pull request by camrynl**
**Reason for Change**:

Adds check to validate CNS pod IP assignment before and after the CNS daemonset is restarted.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 1]
```diff
@@ -165,6 +165,18 @@ steps:
     name: "validatePods"
     displayName: "Validate Pods"
 
+  - script: |
+      echo "validate pod IP assignment and check systemd-networkd restart"
+      kubectl get pod -owide -A
+      make test-validate-state
+      echo "restart CNS"
+      kubectl rollout restart ds azure-cns -n kube-system
```
Comment (vipul-21) [inline .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:173]: Let's add a wait as well. `kubectl rollout status ds azure-cns -n kube-system` to make sure azure-cns has restarted.

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 2]
```diff
@@ -165,6 +165,18 @@ steps:
     name: "validatePods"
     displayName: "Validate Pods"
 
+  - script: |
```
Comment (vipul-21) [inline .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:168]: Should we add cilium agent restarts as well ?

Comment (camrynl) [reply .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:168]: I think that check can be reserved for cilium nightly pipeline runs

**Review by vipul-21 — DISMISSED** [review]

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 3]
```diff
@@ -165,6 +165,19 @@ steps:
...
+      echo "validate pod IP assignment and check systemd-networkd restart"
```
Comment (tamilmani1989) [inline .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:169]: why log message says "systemd-networkd" restart

Comment (vipul-21) [reply .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:169]: The cmd `make test-validate-state` validates the state after restarting systemd-networkd restart for linux as a part of validating state.

**Review by tamilmani1989 — APPROVED** [review]
lgtm

**Pull request merged**